plugins/bazel/README.md

```diff
@@ -0,0 +1,24 @@
+# Bazel Plugin
```

Comment:
Looking at the code, I'm wondering if this just makes sense to be a `version-file-plugin` instead of a Bazel one, since there isn't anything Bazel specific in here.

Reply:
Fair point. Is there an easy way to rename a plugin and its references?

Reply:
Agreed, a Bazel plugin will be interesting though. If that is the purpose of this PR, maybe we should consider how Auto could stamp `package.json` that are output to a `bazel-bin` folder? At the end of the day, Bazel shouldn't look at a version file and have that impact the build, Auto should take the outputs of Bazel and release them appropriately, thoughts?

Reply:
With `auto` would we even need the `VERSION` file? 

Could we just do something like
```sh
#!/usr/bin/env bash
echo STABLE_GIT_COMMIT $(git rev-parse HEAD)
echo STABLE_AUTO_VERSION $(auto version)
```

as the [workspace-status-script](https://bazelbuild.github.io/rules_nodejs/stamping.html#stamping-with-a-workspace-status-script)? Then all of the build-in mechanisms for [stamping a release](https://bazelbuild.github.io/rules_nodejs/stamping.html#substitutions-attribute) would carry over and not be tooling dependent. 

We'd still need a source of truth (and maybe that's where the `VERSION` file comes into play), but could easily also be done using the existing `git tag` plugin. 

`auto shipit` would be interesting though since it'd have to know that `bazel` command to call

Reply:
Yeah that could work, I think we'll just need to make sure auto creates a GitHub release with that still so that the next SemVer calculation will have appropriate context.

Reply:
> Bazel shouldn't look at a version file and have that impact the build

Just to note, the version file approach was heavily influenced by how the publishing Bazel libraries that I've used require the version to be stored:

https://github.com/vaticle/bazel-distribution

So this idea is somewhat interesting. The version file would be needed by the publish targets (if using those rules, which we most likely will be since I already patched those rules to work for our Maven releases), but that wouldn't cause a complete rebuild. Additionally, the idea with this is that whatever publish script you include here would stamp the Bazel build while actually publishing if the version is needed in the actual source.

Reply:
That's a fair callout, assuming that distribution via version file is the expectation within Bazel (admittedly my understanding of non JS/Go publishes is limited by comparison). I have seen recommendations from some [rule maintainers](https://bazelbuild.github.io/rules_nodejs/stamping.html#release-script) to use scripts to correctly query for releasable assets and then substitute the version in via workspace status as @adierkens mentioned. In that case, each rule would just need to inject `STABLE_AUTO_VERSION` appropriately for the version.

Reply:
> assuming that distribution via version file is the expectation within Bazel

Not really the expectation within Bazel, just the distribution rules that I linked above, which include [rules for NPM distribution](https://github.com/vaticle/bazel-distribution#deploy_npm) as well. Even if not using these rules, it might make sense to be able to standardize the repo such that it can be easy to version without Auto, but still work with rules of this style. In addition, it is very easy to stamp with a `VERSION` file:

```
#!/usr/bin/env bash
echo STABLE_VERSION $(cat VERSION)
echo STABLE_GIT_COMMIT $(git rev-parse HEAD)
```

Reply:
This does touch on an issue I did come across while writing this PR. If in a bazel repo you have a script that does the build/stamp/release you don't know what version you need to stamp the built packages with until after auto figures it out as part of `shipit`. So when auto runs the release script as part of whatever kind of release it is (regular, canary, next) the release script needs to restamp the packages _then_ run release. By having a version file and having auto change it, bazel automatically knows it just needs to rerun the stamping step before running its release targets. This flow would prevent bazel from having to rebuild everything from scratch.